mlrun/frameworks/_common/mlrun_interface.py

```diff
@@ -2,19 +2,13 @@
 import functools
 import inspect
 from abc import ABC
-from types import MethodType
-from typing import Any, Dict, Generic, List, Tuple
+from types import FunctionType, MethodType
+from typing import Any, Dict, Generic, List, Tuple, Union
 
-from .utils import MLRunInterfaceableType
+from .utils import Types
```

Comment:
I see your point, but in regard for type hints, I think it's better to let me have these in order to make the code cleaner. A list of lists of integers would look like this: `typing.List[typing.List[int]]`, its kinda ugly. In addition, my custom types are all located in `TFKerasTypes`, so in case I'll use it for hint, for a list of all the models types I'll need to do `typing.List[utils.TFKerasTypes.ModelType]` and the black formatter will crap all over my code.

Reply:
I see, well our main motivation is to avoid circular imports with our own modules and make code more explanatory to what object we reference , sometimes we could have external library with class called `Typing` and also internal class we created called `Types` when taking a first glance on the code it pretty hard to understand if the code reference some external code or internal and when using namespacing this problem disappear, I agree that sometime that could cause some non-intuitive formatting indentation but code wise this could reduce later bugs with referencing or even overriding the wrong objects.

Reply:
Now that I am looking at it, you should also change the `Types` class to be more specific, this is too generic name.

Reply:
Changed both `Utils` and `Types` to `CommonUtils` and `CommonTypes`